include/envoy/common/platform.h

```diff
@@ -248,19 +248,12 @@ struct mmsghdr {
 #endif // __ANDROID_API__ < 24
 #endif // ifdef __ANDROID_API__
 
-#ifdef __linux__
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif
-
 // https://android.googlesource.com/platform/bionic/+/master/docs/status.md
 // ``pthread_getname_np`` is introduced in API 26
-#ifdef __ANDROID_API__
-#if __ANDROID_API__ > 26
-#define SUPPORTS_PTHREAD_GETNAME_NP 1
-#endif // __ANDROID_API__ > 26
-#endif // ifdef __ANDROID_API__
-
-// Ensure `SUPPORTS_PTHREAD_GETNAME_NP` is set
-#ifndef SUPPORTS_PTHREAD_GETNAME_NP
-#define SUPPORTS_PTHREAD_GETNAME_NP 0
-#endif
+#if defined(__ANDROID_API__)
+#if __ANDROID_API__ >= 26
+#define SUPPORTS_PTHREAD_GETNAME_NP
+#endif // __ANDROID_API__ >= 26
+#elif defined(__linux__)
+#define SUPPORTS_PTHREAD_GETNAME_NP
+#endif // defined(__ANDROID_API__)
```

Comment:
@jmarantz requested on the other PR that these be set to numerical values for (potential) boolean logic comparisons.

Reply:
I opted for basic `#define`s here to keep the logic in this block simple. Using numeric-backed definitions would make this significantly more complicated and less readable. If people feel strongly I can change it.

Reply:
I don't think that would change the logic at all; you'd just be adding a ` 1` to the end of a couple of lines.

Reply:
We would also need to either add 2 `#else` cases to define it as zero in all branches, or add a separate block with an `#ifndef`, right?

```
#if defined(__ANDROID_API__)
#if __ANDROID_API__ >= 26
#define SUPPORTS_PTHREAD_GETNAME_NP 1
#else
#define SUPPORTS_PTHREAD_GETNAME_NP 0
#endif // __ANDROID_API__ >= 26
#elif defined(__linux__)
#define SUPPORTS_PTHREAD_GETNAME_NP 1
#else
#define SUPPORTS_PTHREAD_GETNAME_NP 0
#endif // defined(__ANDROID_API__)
```

Reply:
i see. just define it to 0 at the start and undef/redefine it to 1 as needed?

Reply:
Updated.